File: 1.two-sum/solution.py
#O(n^2)
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        length = len(nums)
        for i in range(length-1):
            first = nums[i]
            expected = target - first
            for j in range(i+1, length):
                if nums[j] == expected:
                    return [i, j]

File: 1.two-sum/solution.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i = 0; i < nums.length; i++) {
            int compl = target - nums[i];
            Integer complIndex = map.get(compl);
            if (complIndex != null) {
                return new int[]{complIndex, i};
            }
            map.put(nums[i], i);
        }
        return null;
    }
}